core/src/serde.rs

```diff
@@ -0,0 +1,39 @@
+use proc_macro2::Span;
+use syn::Ident;
+
+use crate::util::IdentString;
+
+impl<'de> serde::Deserialize<'de> for IdentString {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        deserializer.deserialize_str(IdentStringVisitor)
+    }
+}
+
+struct IdentStringVisitor;
+
+impl<'de> serde::de::Visitor<'de> for IdentStringVisitor {
+    type Value = IdentString;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(formatter, "a valid ident")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(IdentString::new(Ident::new(v, Span::call_site())))
+    }
+}
+
+impl serde::Serialize for IdentString {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.as_str())
+    }
```

Comment:
It would be nice to add some tests

Reply:
I actually thought about that. This would pull in at least `serde_json` or `serde_yaml` as a dev dependency. If we are fine with that, I can add a few unit tests.

Reply:
I'm on the fence about this - I would like to see some tests, but I am worried about the impact on the build and test time.

Reply:
I can add some small tests and then we can see how the test times are impacted. If you are not satisfied, I can easily revert the commit.

Reply:
Added in e39a4bb.